### PR TITLE
switch bevy_image to use wgpu-types wherever possible instead of wgpu

### PR DESCRIPTION
**Pull request by atlv24**
# Objective

- dont depend on wgpu if we dont have to

## Solution

- works towards this, but doesnt fully accomplish it. the remaining types stopping us from doing this need to be moved upstream, i will PR this

## Testing

- 3d_scene runs

### CODE REVIEW

**Comment by atlv24** [comment]
Those 3 are the only types which need moving. Once these merge we will be able to only pull in wgpu-types for bevy_image and bevy_mesh (and soon bevy_camera)

**Comment by atlv24** [comment]
All PRs have been merged, this will be good to go next wgpu release.

File: crates/bevy_image/Cargo.toml [hunk 1]
```diff
@@ -49,6 +49,7 @@ image = { version = "0.25.2", default-features = false }
 # misc
 bitflags = { version = "2.3", features = ["serde"] }
 bytemuck = { version = "1.5" }
+wgpu-types = { version = "23", default-features = false }
 wgpu = { version = "23.0.1", default-features = false }
```
Comment (kristoff3r) [inline crates/bevy_image/Cargo.toml:53]: Reminder to remember to remove this

**Review by kristoff3r — APPROVED** [review]

File: crates/bevy_image/Cargo.toml (hunk 1 above)
Comment (atlv24) [reply crates/bevy_image/Cargo.toml:53]: can't remove it this PR, but once we upgrade wgpu we can

File: crates/bevy_image/Cargo.toml [hunk 2]
```diff
@@ -49,6 +49,7 @@ image = { version = "0.25.2", default-features = false }
 # misc
 bitflags = { version = "2.3", features = ["serde"] }
 bytemuck = { version = "1.5" }
+wgpu-types = { version = "23", default-features = false }
```
Comment (mockersf) [inline crates/bevy_image/Cargo.toml:52]:
```suggestion
wgpu-types = { version = "23", default-features = false }
# TODO: remove dependency on wgpu once https://github.com/gfx-rs/wgpu/pull/6648, 6649 and 6650 have been released
```

**Review by mockersf — APPROVED** [review]

**Pull request merged**